### PR TITLE
init push or pytest conversion of override_database for bigquery

### LINKED ISSUE #164: [CT-536] convert override_database to pytest method
### Describe the feature
Convert Integration test Override_database into pytest method

### Who will this benefit?
Everyone, pytest structure is more common therefore more easy to understand from.

### Are you interested in contributing this feature?
yep

### PR DESCRIPTION
**Pull request by McKnight-42**
resolves #164

### Description

Converting the BigQuery implementation of override_database to the new pytest format

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [ ] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-bigquery next" section.

### CODE REVIEW

**Comment by McKnight-42** [comment]
I have ran the conversion script and started playing around with the script to try and get it to work and have a question about weather or not there are additonal changes we might have to make around getting the env variables since these aren't just replacing values from dbt-core confftest.py

current log from when I run a test:

```

self = <test_override_database.TestModelOverride object at 0x7f9787cd4e90>

    def test_bigquery_database_override(self):
>       self.run_database_override()

tests/functional/adapter/test_override_database/test_override_database.py:53:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/functional/adapter/test_override_database/test_override_database.py:41: in run_database_override
    run_dbt(['seed'])
../dbt-core/core/dbt/tests/util.py:70: in run_dbt
    res, success = handle_and_check(args)
../dbt-core/core/dbt/main.py:191: in handle_and_check
    task, res = run_from_args(parsed)
../dbt-core/core/dbt/main.py:218: in run_from_args
    task = parsed.cls.from_args(args=parsed)
../dbt-core/core/dbt/task/base.py:184: in from_args
    move_to_nearest_project_dir(args)
../dbt-core/core/dbt/task/base.py:171: in move_to_nearest_project_dir
    nearest_project_dir = get_nearest_project_dir(args)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = Namespace(cache_selected_only=None, cls=<class 'dbt.task.seed.SeedTask'>, debug=None, defer=None, event_buffer_size=No...lors=None, use_experimental_parser=None, vars='{}', version_check=None, warn_error=None, which='seed', write_json=None)

    def get_nearest_project_dir(args):
        # If the user provides an explicit project directory, use that
        # but don't look at parent directories.
        if args.project_dir:
            project_file = os.path.join(args.project_dir, "dbt_project.yml")
            if os.path.exists(project_file):
                return args.project_dir
            else:
                raise dbt.exceptions.RuntimeException(
                    "fatal: Invalid --project-dir flag. Not a dbt project. "
                    "Missing dbt_project.yml file"
                )

        root_path = os.path.abspath(os.sep)
        cwd = os.getcwd()

        while cwd != root_path:
            project_file = os.path.join(cwd, "dbt_project.yml")
            if os.path.exists(project_file):
                return cwd
            cwd = os.path.dirname(cwd)

        raise dbt.exceptions.RuntimeException(
>           "fatal: Not a dbt project (or any of the parent directories). "
            "Missing dbt_project.yml file"
        )
E       dbt.exceptions.RuntimeException: Runtime Error
E         fatal: Not a dbt project (or any of the parent directories). Missing dbt_project.yml file

../dbt-core/core/dbt/task/base.py:165: RuntimeException
----------------------------------------------------------------------------------------------------------------------------- Captured stdout call -----------------------------------------------------------------------------------------------------------------------------


Invoking dbt with ['seed']
=========================================================================================================================== short test summary info ============================================================================================================================
FAILED tests/functional/adapter/test_override_database/test_override_database.py::TestModelOverride::test_bigquery_database_override - dbt.exceptions.RuntimeException: Runtime Error
```

File: tests/functional/adapter/test_override_database/test_override_database.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            }
+        }
+
+
+class TestModelOverrideBigQuery(BaseOverrideDatabase):
+    def run_database_override(self, project):
+        run_dbt(["seed"])
+        assert len(run_dbt(["run"])) == 4
+        check_relations_equal_with_relations(project.adapter, [
+            project.adapter.Relation.create(database=os.getenv("BIGQUERY_TEST_DATABASE"), schema=project.test_schema, identifier="seed"),
```
Comment (jtcohen6) [inline tests/functional/adapter/test_override_database/test_override_database.py:40]: We could either:
- Replace `database=os.getenv("BIGQUERY_TEST_DATABASE")` with `database=project.database`
- Remove the `database` argument entirely, and fix a v1.1 regression (https://github.com/dbt-labs/dbt-bigquery/issues/180) by restoring support for using default `database`/`schema` when unspecified in `get_columns_in_relation`

File: tests/functional/adapter/test_override_database/test_override_database.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        return {
+            "config-version": 2,
+            "vars": {
+                "alternate_db": os.getenv("BIGQUERY_TEST_ALT_DATABASE"),
+            },
+            "models": {
+                "database": os.getenv("BIGQUERY_TEST_ALT_DATABASE"),
+                "test": {
+                    "subfolder": {
+                        "database": os.getenv("BIGQUERY_TEST_DATABASE")
```
Comment (jtcohen6) [inline tests/functional/adapter/test_override_database/test_override_database.py:81]: You could replace this env var pretty straightforwardly with either of the following.

More like what users would actually write in their projects:
```suggestion
                        "database": "{{ target.database }}"
```

More like what this is doing currently:
```suggestion
                        "database": project.database
```

**Review by jtcohen6 — COMMENTED** [review]
We actually don't use `BIGQUERY_TEST_DATABASE` anywhere else. In the `service_account` profile, we use `project_id`, pulled out of the service account JSON. For `oauth` connections, we use the default configured via `gcloud config`. I think we could swing this test without that env var, too—but I'm open to leaving it in if you think it's better that way.

**Organization of test cases:** I wouldn't expect this test case to land in `tests/functional/adapter`, I would only expect that to be test cases that are pulled from `dbt-tests-adapter`. I might expect this test case to land in `tests/functional/<topic_area>/test_override_database/...`

Finally: We can also delete the legacy test (`tests/integration/override_database_test`), right?

**Comment by McKnight-42** [comment]
> **Env var:** We actually don't use `BIGQUERY_TEST_DATABASE` anywhere else. In the `service_account` profile, we use `project_id`, pulled out of the service account JSON. For `oauth` connections, we use the default configured via `gcloud config`. I think we could swing this test without that env var, too—but I'm open to leaving it in if you think it's better that way.
> 
> **Organization of test cases:** I wouldn't expect this test case to land in `tests/functional/adapter`, I would only expect that to be test cases that are pulled from `dbt-tests-adapter`. I might expect this test case to land in `tests/functional/<topic_area>/test_override_database/...`
> 
> **Finally:** We can also delete the legacy test ([`tests/integration/override_database_test`](https://github.com/dbt-labs/dbt-bigquery/tree/main/tests/integration/override_database_test)), right?

I'm find removing it if we were only using it to test this use case, I know we also have the dbt-dev account and the alt account still so that covers these cases well for ci/cd and we still have the no access account to test against for that use case. would just need to make sure we remove it from bigquery eventually.

File: tests/functional/adapter/test_override_database/test_override_database.py (hunk 2 above)
Comment (McKnight-42) [reply tests/functional/adapter/test_override_database/test_override_database.py:81]: I think the target.database works more as that part of the function doesn't generally have access to project and when i tried to tap it it gave recursive dependency issues.

**Review by ChenyuLInx — APPROVED** [review]
LGTM! Glad to see all things related to compare different database object resolved!

File: dbt/adapters/bigquery/connections.py [hunk 3]
```diff
@@ -555,6 +555,9 @@ def table_ref(database, schema, table_name):
     def get_bq_table(self, database, schema, identifier):
         """Get a bigquery table for a schema/model."""
         conn = self.get_thread_connection()
+        # backwards compatibility: fill in with defaults if not specified
+        database = database or conn.credentials.database
+        schema = schema or conn.credentials.schema
```
Comment (jtcohen6) [inline dbt/adapters/bigquery/connections.py:560]: To discuss: Backporting this piece to `1.1.latest`, to resolve https://github.com/dbt-labs/dbt-bigquery/issues/180. Let's just add a test case for the regression case as well. It's really just calling `adapter.get_columns_in_relation` on a `Relation` that's missing `database`/`schema`

Comment (McKnight-42) [reply dbt/adapters/bigquery/connections.py:560]: Do we want this test in the test_bigquery_adapter section using the magic mock?

Comment (McKnight-42) [reply dbt/adapters/bigquery/connections.py:560]: or does this fall under more of a integration test?

**Pull request merged**